### PR TITLE
Activity Log: introduce a month selector

### LINKED ISSUE #14665: Activity Log: Date Select component
When the activity spans across more than one month, we add a month select component to allow easier navigation. This component can be expanded in time to allow searching/filtering.

image

### PR DESCRIPTION
**Pull request by eliorivero**
Fixes #14665

This will display a month selector on the top of the page. When the arrows are clicked, the events displayed will change based on the month selected. Will update timestamps in data so it displays events for June, May, April, and January.

monthselector
(for demo purposes I deleted the banners)

### CODE REVIEW

File: client/my-sites/stats/stats-date-picker/index.jsx [hunk 1]
```diff
@@ -125,20 +127,32 @@ class StatsDatePicker extends Component {
...
-					<span className="period">
-						<span className="date">{ isSummarizeQuery ? this.dateForSummarize() : this.dateForDisplay() }</span>
-					</span>
-				)
-			},
-			context: 'Stats: Main stats page heading',
-			comment: 'Example: "Stats for December 7", "Stats for December 8 - December 14", "Stats for December", "Stats for 2014"'
-		} );
+		const sectionTitle = isActivity
+			? translate( 'Activity for {{period/}}', {
```
Comment (a8ci18n) [inline client/my-sites/stats/stats-date-picker/index.jsx]: :information_source: This **context** is really long. Are you sure you don't want to use a translator comment instead?

File: client/my-sites/stats/activity-log/index.jsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import React, { Component } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
 import { groupBy, map } from 'lodash';
+import moment from 'moment';
```
Comment (timmyc) [inline client/my-sites/stats/activity-log/index.jsx]: The `localize` HoC will inject a `moment` instance as a [prop](https://github.com/Automattic/i18n-calypso#moment-method) so we don't need to import moment here

File: client/my-sites/stats/activity-log/index.jsx [hunk 3]
```diff
@@ -275,9 +279,9 @@ class ActivityLog extends Component {
 	render() {
 		const {
 			isJetpack,
-			moment,
 			siteId,
 			slug,
+			startDate,
```
Comment (timmyc) [inline client/my-sites/stats/activity-log/index.jsx]: if we add `moment` here, then all will work perfectly with the `i18n-moment`

File: client/my-sites/stats/activity-log/index.jsx [hunk 4]
```diff
@@ -4,7 +4,8 @@
 import React, { Component, PropTypes } from 'react';
 import { connect } from 'react-redux';
 import { localize } from 'i18n-calypso';
-import { groupBy, map, get } from 'lodash';
+import { groupBy, map, get, filter } from 'lodash';
+import moment from 'moment';
```
Comment (dereksmart) [inline client/my-sites/stats/activity-log/index.jsx]: As Timmy said, we don't need to import this, as we can get it from the Localize props

Comment (eliorivero) [inline client/my-sites/stats/activity-log/index.jsx]: Sorry, this slipped through again when I rebased yesterday. Fixed now.

File: client/my-sites/stats/stats-date-picker/index.jsx [hunk 5]
```diff
@@ -125,20 +127,32 @@ class StatsDatePicker extends Component {
...
+				components: {
+					period: (
+						<span className="period">
+							<span className="date">{ isSummarizeQuery ? this.dateForSummarize() : this.dateForDisplay() }</span>
+						</span>
+					)
+				},
+				context: 'Activity Log: Main activity section heading',
```
Comment (seear) [inline client/my-sites/stats/stats-date-picker/index.jsx]: It is likely that we don't need `context` here. [`context`](https://codex.wordpress.org/I18n_for_WordPress_Developers#Disambiguation_by_context) is only necessary when two phrases that are the same in english may have different meanings in some languages.

Information about the string can go in `comment`.

Comment (eliorivero) [inline client/my-sites/stats/stats-date-picker/index.jsx]: Updated to remove context.

**Comment by eliorivero** [comment]
Closing this in favor of #15217 since I broke this branch.

**Pull request merged**